packages/overture-schema-system/src/overture/schema/system/metadata.py

```diff
@@ -0,0 +1,319 @@
+from collections.abc import (
```

Comment:
The metadata is a formalization and generalization of the `model_constraint` metadata introduced in #400 track the model constraints applied on a model for code generation support.